src/pybind/ceph_volume_client.py

```diff
@@ -306,7 +306,7 @@ def recover(self):
             # Identify auth IDs from auth meta filenames. The auth meta files
             # are named as, "$<auth_id><meta filename extension>"
             regex = "^\$(.*){0}$".format(re.escape(META_FILE_EXT))
-            match = re.search(regex, d.d_name)
+            match = re.search(regex, d.d_name.decode())
```

Comment:
@rishabh-d-dave 
Let's explicitly mention that the encoding type is 'utf-8'. It's the default in py3, but not in py2, which is OK for octopus. We'll want to backport this PR to nautilus for which the library should be py2 compatible.

Reply:
Done.

Reply:
it should be, d.d_name.decode(encoding='utf-8')

Reply:
isn't that already implied?

```
>>> a
b'abcd'
>>> type(a)
<class 'bytes'>
>>> type(a.decode())
<class 'str'>
>>> type(a.decode(encoding='utf-8'))
<class 'str'>
```

Reply:
as I said earlier it's 'utf-8' by default in python 3 but not in python 2. so when we backport to nautilus this would be helpful.

Reply:
we have already have a comment to help.